include/caffe/filler.hpp

```diff
@@ -60,6 +61,40 @@ class UniformFiller : public Filler<Dtype> {
   }
 };
 
+/// @brief Fills a Blob with uniformly distributed values @f$ x\sim U(a, b) @f$.
+//         RNG is only called once, and its output is stored in
+//         UniformStaticFillerdata. Followup calls to the Filler function
+//         will bypass RNG.
+template <typename Dtype>
+class UniformStaticFiller : public Filler<Dtype> {
+ public:
+  explicit UniformStaticFiller(const FillerParameter& param)
+      : Filler<Dtype>(param) {
+        uniform_static_filler_flag_ = false;
+      }
+  virtual void Fill(Blob<Dtype>* blob) {
+    Dtype* data = blob->mutable_cpu_data();
+    const int blob_count = blob->count();
+    CHECK(blob_count);
+
+    if (uniform_static_filler_flag_ == false) {
+      uniform_static_filler_data_.resize(blob_count);
+      caffe_rng_uniform(blob_count, Dtype(this->filler_param_.min()),
+          Dtype(this->filler_param_.max()),
+          &uniform_static_filler_data_.front());
+      uniform_static_filler_flag_ = true;
```

Comment:
That is right. So the reason this was requested was for benchmarking frameworks without I/O or RNG bottlenecks. One can benchmark these bottlenecks separately, but the idea is to measure the forward, backward and update times and the scalability across multiple processes for different networks. For cases where a new random number is needed, the other layer types can be used.